test/test_torch.py

```diff
@@ -6473,6 +6473,25 @@ def test_isfinite_isinf_isnan_complex(self, device, dtype):
         self.compare_with_numpy(torch.isinf, np.isinf, vals, device, dtype)
         self.compare_with_numpy(torch.isnan, np.isnan, vals, device, dtype)
 
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(torch.complex64, torch.complex128)
+    def test_isreal_complex(self, device, dtype):
+        vals = (1, 1 + 1j, 2 + 0j, 3j, 2 - 1j, 2 - 0j)
+        self.compare_with_numpy(torch.isreal, np.isreal, vals, device, dtype)
+
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_isreal_noncomplex(self, device, dtype):
+        if dtype is not torch.bfloat16:  # numpy supports all dtypes torch does except bfloat16
+            vals = (1, 2, 3)
+            self.compare_with_numpy(torch.isreal, np.isreal, vals, device, dtype)
+
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(torch.float32)
+    def test_isreal_nan(self, device, dtype):
```

Comment:
I think you can remove this test and add `float('nan')` to the previous test if it stops comparing with NumPy.

Reply:
I separated this to declutter the noncomplex test, since we'd have to separate int from floats using NaN there.